File: appli-android/app/src/main/java/com/example/sportapp/core/network/QuoteApi.kt
package com.example.sportapp.core.network

import com.example.sportapp.core.data.model.Quote
import retrofit2.http.*

interface QuoteApi {

    @GET("quotes")
    suspend fun getAll(): List<Quote>

    @GET("quotes/{uuid}")
    suspend fun getByUUID(@Path("uuid") uuid: String): Quote

    @PUT("quotes/{uuid}")
    suspend fun upsert(@Path("uuid") uuid: String, @Body item: Quote)

    @PUT("quotes/bulk")
    suspend fun upsertAll(@Body items: List<Quote>)

    @DELETE("quotes/{uuid}")
    suspend fun delete(@Path("uuid") uuid: String)
}

File: appli-web/src/app/core/api/quote-api.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { Quote } from '@core/models/quote.model';
import { API_BASE_URL } from './api.config';

@Injectable({ providedIn: 'root' })
export class QuoteApi {
  private readonly http = inject(HttpClient);
  private readonly base = `${API_BASE_URL}/quotes`;

  getAll(): Observable<Quote[]> {
    return this.http.get<Quote[]>(this.base);
  }
  upsert(q: Quote): Observable<Quote> {
    return this.http.put<Quote>(`${this.base}/${q.uuid}`, {
      uuid: q.uuid,
      text: q.text,
      author: q.author ?? null,
      updatedAt: q.updatedAt,
    });
  }
  /** Upsert groupé `PUT /quotes/bulk` (miroir upsertAll Android) — 1 requête pour N rows. */
  upsertAll(items: Quote[]): Observable<Quote[]> {
    return this.http.put<Quote[]>(
      `${this.base}/bulk`,
      items.map((q) => ({
        uuid: q.uuid,
        text: q.text,
        author: q.author ?? null,
        updatedAt: q.updatedAt,
      })),
    );
  }
  delete(uuid: string): Observable<unknown> {
    return this.http.delete(`${this.base}/${uuid}`);
  }
}
